src/d3a/models/strategy/external_strategies/load.py

```diff
@@ -345,6 +355,82 @@ def _list_bids_aggregator(self, arguments):
+                  'update_interval', 'initial_buying_rate', 'final_buying_rate',
+                  'balancing_energy_ratio', 'use_market_maker_rate')
+
+    def __init__(self, power_forecast_W: float = 0,
+                 fit_to_limit=True, energy_rate_increase_per_update=None,
+                 update_interval=None,
+                 initial_buying_rate: Union[float, dict, str] =
+                 ConstSettings.LoadSettings.BUYING_RATE_RANGE.initial,
+                 final_buying_rate: Union[float, dict, str] =
+                 ConstSettings.LoadSettings.BUYING_RATE_RANGE.final,
+                 balancing_energy_ratio: tuple =
+                 (ConstSettings.BalancingSettings.OFFER_DEMAND_RATIO,
+                  ConstSettings.BalancingSettings.OFFER_SUPPLY_RATIO),
+                 use_market_maker_rate: bool = False):
+        """
+        Constructor of LoadForecastStrategy
+        :param power_forecast_W: forecast for the next market slot
+        """
+        if update_interval is None:
+            update_interval = \
+                duration(minutes=ConstSettings.GeneralSettings.DEFAULT_UPDATE_INTERVAL)
+
+        super().__init__(daily_load_profile=None,
+                         fit_to_limit=fit_to_limit,
+                         energy_rate_increase_per_update=energy_rate_increase_per_update,
+                         update_interval=update_interval,
+                         final_buying_rate=final_buying_rate,
+                         initial_buying_rate=initial_buying_rate,
+                         balancing_energy_ratio=balancing_energy_ratio,
+                         use_market_maker_rate=use_market_maker_rate)
+
+        self.power_forecast_buffer_W = power_forecast_W
+
+    @property
+    def channel_dict(self):
+        return {**super().channel_dict,
+                f'{self.channel_prefix}/set_power_forecast': self._set_power_forecast}
+
+    def _incoming_commands_callback_selection(self, req):
+        if req.request_type == "bid":
+            self._bid_impl(req.arguments, req.response_channel)
+        elif req.request_type == "delete_bid":
+            self._delete_bid_impl(req.arguments, req.response_channel)
+        elif req.request_type == "list_bids":
+            self._list_bids_impl(req.arguments, req.response_channel)
+        elif req.request_type == "device_info":
+            self._device_info_impl(req.arguments, req.response_channel)
+        elif req.request_type == "set_power_forecast":
+            self._set_power_forecast_impl(req.arguments, req.response_channel)
+        else:
+            assert False, f"Incorrect incoming request name: {req}"
+
+    def event_market_cycle(self):
+        self.update_energy_forecast()
+        super().event_market_cycle()
+
+    def event_activate_energy(self):
+        self.update_energy_forecast()
+
+    def update_energy_forecast(self):
+        self._initiate_hrs_per_day()
```

Comment:
Why do you need to call this again ? This should be called by the _update_energy_requirement (essentially the first time that the energy_requirement is populated), therefore there should be no need to be called from here too.

Reply:
I was not set, because the `event_activate_energy` is overwritten here. But you are right, it should live in there. Moving it.